src/FixedDataTable.js

```diff
@@ -639,7 +640,8 @@ var FixedDataTable = React.createClass({
   },
 
   _renderRows(/*number*/ offsetTop) /*object*/ {
-    var state = this.state;
+    let state = updateRowHeights(this.state, this.props);
+    state = updateVisibleRows(state);
```

Comment:
Woh, we definitely don't want to call this here. They are expensive operations. 
We need to track what went wrong in the reducer and add correct paths there.